### ISSUE TITLE
Asynchronous component

### ISSUE THREAD

**Issue by yhatt** [speee/jsx-slack#153]
I discovered jsx-slack v2 can use asynchronous component as following:

```jsx
const AsyncComponent = async ({ userId }) => {
  // Fetch some data
  const name = await getUserName({ id: userId })

  return <Section>Hello, {name}!</Section>
}

;(async () => {
  console.log(
    <Blocks>
      <Context>Asynchronous component example</Context>
      <Divider />
      {await <AsyncComponent userId={123} />}
    </Blocks>
  )
})()
```

TypeScript requires to cast the component into `any` type.

This usage was found out by chance and it's **out of our specification**, but may have a worth to write down to the documentation as one of the useful way to use async functions.

**Comment by varadiistvan** [comment]
Hey! We're using jsx-slack in our project, and was wondering if this was possible on current version? The workaround we found for now is calling components as functions, but that's kind of ugly to do

**Comment by yhatt** [comment]
Yes, this code is still working in JavaScript with the current version of jsx-slack (v6), without no changes. 😎
